malaria-ios/Managers/MedicineHelpers/MedicineManager.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+class MedicineManager{
+    static let sharedInstance = MedicineManager()
+    
+    
+    
+    /// Setups a new pill and sets it as default
+    ///
+    /// If there is already a pill registed as default, it will no longer be the default pill
+    ///
+    /// :param: `Medicine.Pill`: the pill
+    /// :param: `NSDate`: fireDate
+    func setup(medicine : Medicine.Pill, fireDate: NSDate){
+        //unregister previous medicines
+        if let currentMed = getCurrentMedicine(){
+            currentMed.notificationManager.unsheduleNotification()
+        }
+        
+        Logger.Info("Storing new medicine")
+        registerNewMedicine(medicine)
+        setCurrentPill(medicine)
+        
+        Logger.Info("Setting up notification")
+        let newMed = getCurrentMedicine()!
+        
+        newMed.notificationManager.scheduleNotification(fireDate)
+
+        UserSettingsManager.setBool(UserSetting.DidConfiguredMedicine, true)
+        UserSettingsManager.syncronize()
```

Comment:
Yeah during the course of the application cycle, it's saved frequently when modified.  Only calling on exit is a good strategy, some would argue that's not even necessary because how quickly that's saved - this is a minor point however.